src/System.Memory/ref/System.Memory.cs

```diff
@@ -246,6 +266,47 @@ public abstract partial class OwnedMemory<T> : System.Buffers.IRetainable, Syste
         public abstract void Retain();
         protected internal abstract bool TryGetArray(out System.ArraySegment<T> arraySegment);
     }
+    public static partial class ReadOnlyBufferExtensions
```

Comment:
I thought you said you will add it to MemoryExtensions?

Reply:
Yes, but then I've seen that MemoryExtensions are in `System` namespace and `ROB` is in `System.Buffers`?: https://github.com/dotnet/corefxlab/pull/2088#discussion_r165149130

Reply:
sorry, I missed the previous reply.

Why do you care that they are in different namespaces?

Reply:
Don't we want to put extension class in the same namespace that class being extended?

Reply:
I don't think it's strange. The only thing that makes me pause if what if we want to move MemoryExtensions lower than '''ReadOnlyBuffer<T>'''. @ahsonkhan, @jan, do you think we will ever want to do it? If yes, then maybe we do want a separate extension holder for ReadOnlyBuffer, or simply make the members instance members.

Reply:
Define "lower".
We do plan to duplicate MemoryExtensions down in CoreLib and typeforward to it for fast span.

> If yes, then maybe we do want a separate extension holder for ReadOnlyBuffer, or simply make the members instance members.

I think we should keep these separate (or make them instance members), regardless.